src/main/java/ch/njol/skript/conditions/CondMatches.java

```diff
@@ -0,0 +1,80 @@
+/**
+ * This file is part of Skript.
+ *
+ * Skript is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Skript is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Skript.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ *
+ * Copyright 2011-2017 Peter Güttinger and contributors
+ */
+
+package ch.njol.skript.conditions;
+
+import java.util.Arrays;
+import java.util.regex.Pattern;
+
+import org.bukkit.event.Event;
+import org.eclipse.jdt.annotation.Nullable;
+
+import ch.njol.skript.Skript;
+import ch.njol.skript.doc.Description;
+import ch.njol.skript.doc.Examples;
+import ch.njol.skript.doc.Name;
+import ch.njol.skript.doc.Since;
+import ch.njol.skript.lang.Condition;
+import ch.njol.skript.lang.Expression;
+import ch.njol.skript.lang.SkriptParser;
+import ch.njol.util.Kleenean;
+
+@Name("Matches")
+@Description("Test whether strings match defined regular expressions")
+@Examples({"on chat:",
+	"\tif message matches \"\\d\":",
+	"\t\tsend \"Message contains a digit!\""})
+@Since("INSERT VERSION")
+public class CondMatches extends Condition {
+	
+	static {
+		Skript.registerCondition(CondMatches.class, "%strings% (1¦match[es]|2¦does(n't| not) match) %strings%");
+	}
+	
+	@SuppressWarnings("null")
+	Expression<String> strings;
+	@SuppressWarnings("null")
+	Expression<String> regex;
+	
+	@Override
+	@SuppressWarnings({"unchecked", "null"})
+	public boolean init(Expression<?>[] exprs, int matchedPattern, Kleenean isDelayed, SkriptParser.ParseResult parseResult) {
+		strings = (Expression<String>) exprs[0];
+		regex = (Expression<String>) exprs[1];
+		setNegated(parseResult.mark == 1);
+		return true;
+	}
+	
+	@Override
+	public boolean check(Event e) {
+		String[] txt1 = strings.getArray(e);
+		String[] patterns = regex.getArray(e);
+		if (txt1.length < 1 || patterns.length < 1) return false;
+		return Arrays.stream(txt1)
+			.allMatch((str) -> Arrays.stream(patterns)
+				.allMatch((pattern) -> (Pattern.compile(pattern).matcher(str).find()))) == isNegated();
```

Comment:
Are you sure you want to use `Matcher#find()` here and not `Matcher#matches()`?
Also, you're using `allMatch` for both of these, but I think that should be dependant on whether it's an `and` list or an `or` list (see https://github.com/SkriptLang/Skript/blob/master/src/main/java/ch/njol/skript/lang/Expression.java#L76)

Reply:
Yeah, I already asked in the skript-chat discord server whether or not skript would automatically do something based on whether or not and or 'or' were used, and since nobody answered I didn't know

Reply:
Thanks

Reply:
Also I wanted to use Matcher.find() because .matches() matches the entire string and I don't want it to have to match that. Users can just use ^ and $ to match the beginning and end

Reply:
> Also I wanted to use Matcher.find() because .matches() matches the entire string and I don't want it to have to match that. Users can just use ^ and $ to match the beginning and end

In that case you should probably think about the syntax again, because `%string% matches %string%` implies that one string matches some regex pattern, AKA the whole string matches some regex pattern, not a part of the given string (`%string` partially matches %string%` maybe, and if you're going for something like that, you should make partially optional to either match the full string or part of it)